stdlib/option.mli

```diff
@@ -82,3 +88,23 @@ val to_list : 'a option -> 'a list
 val to_seq : 'a option -> 'a Seq.t
 (** [to_seq o] is [o] as a sequence. [None] is the empty sequence and
     [Some v] is the singleton sequence containing [v]. *)
+
+(** {1:syntax Syntax} *)
+
+(** Binding operators.
+
+    @since 5.4 *)
+module Syntax : sig
+
+  val ( let* ) : 'a option -> ('a -> 'b option) -> 'b option
+  (** [( let* )] is {!Option.bind}. *)
```

Comment:
I would prefer if API docs for binding operators would show the syntax in use, or at least if they would refer explicitly to a chapter in the manual that explains this. Currently the documentation is obscure if you don't already know binding operators, and the syntax is not self-discoverable.

Reply:
I added a pointer to the relevant manual section. I did the same for `Result.Syntax`. Examples would be nice, but it will probably be a more sizeable task and better taken up on its own. Please yell if you are not happy with the current state of the PR. Otherwise I am planning to merge by end of day.